SRC/plotTableFromMarkdown.py

```diff
@@ -1,37 +1,109 @@
 import sys
+import os
 import re
 import matplotlib.pyplot as plt
+from tkinter import filedialog
+import tkinter as tk
 
-if len(sys.argv) == 1:
-	lines = sys.stdin.readlines()
-else:
-	with open(sys.argv[1]) as file:
-		lines = file.readlines()
-
-x = []
-y = []
-xLabel = ""
-yLabel = ""
-
-for line in lines:
-	if re.match( "\\|\s*---+\s*\\|\s*---+\s*", line):
-		continue
-
-	matchObj = re.match( "\\|\s*(\.?\d+\.?\d*)\s*\\|\s*(\.?\d+\.?\d*)\s*", line)
-	if matchObj:
-		value = float(matchObj.group(1))
-		x.append(value)
-		value = float(matchObj.group(2))
-		y.append(value)
-		continue
-
-	matchObj = re.match( "\\|\s(.+?)\s\\|\s(.+?)\s\\|", line)
-	if matchObj and xLabel == "" and yLabel == "":
-		xLabel = matchObj.group(1)
-		yLabel = matchObj.group(2)
-		continue
-
-plt.plot(x,y,'ko')
-plt.xlabel(xLabel)
-plt.ylabel(yLabel)
-plt.show()
+
+class Plotter:
+    def __init__(self):
+        root = tk.Tk()
+        root.withdraw()
+        self.acceptedExtensions = [".md"]
+        self.compatible = 0
+        self.filePath = ""
+        self.fileExtension = ""
+        self.lines = ''
+
+        self.askUserOpenFile()
+        self.findFileExtension()
+        self.printFileInfo()
+        self.extensionVerify()
+        self.extentionHandle()
+
+        self.dataPlot()
+
```

Comment:
I think it would be cleaner to only init variables without calling functions.
You would then call, let's say, `DataHandler().plotDataFromFile()`